### ISSUE TITLE
Lockfile and scripts after isolate

### ISSUE THREAD

**Issue by torbjorn-kvist** [0x80/mono-ts#9]
Love the monorepo setup, been testing it out during the week and have some questions. Not sure if this is the best place for it. If not, please suggest where I should ask :) 

1. Been testing out the mono-ts example and noticed that the pnpm-lock.yaml for `services/fns` contains `next` dependencies, even when the package.json contains no such dependencies. Tried `pnpm dedupe` inside the isolate file and it shrinks by quite a bit. 

2. Is there anyway to run scripts after the isolate phase when using `firebase-tools-isolate`? Have use-case to copy some assets that I don't want isolate to "mess" with. It works with `fns` example, by specifying it after the `isolate` step in firebase.json but I saw that you wrote that example is not a recommend way of doing it.

**Comment by 0x80** [comment]
Not sure about the dependencies. Thanks for bringing it under my attention. I will look into it, but I'm quite swamped with stuff at the moment.

Using isolate directly is only not recommended because its cumbersome to run the emulators like that. You do not get hot module reloading like you normally have with running the firebase emulators directly. If you don't care then there's no harm in triggering isolate yourself before deployment.

What are you worried about concerning your assets? The isolation process just copies files and touches package.json files and lockfiles. The files that get moved are not mutated.

If you were to include the assets in the "files" field of your manifest, they will basically just get zipped and unzipped, and that's it.

**Comment by torbjorn-kvist** [comment]
No problem! Take your time, it's just something that I noticed when testing around. Saw that the uploaded bundle was bigger then I expected and its not a blocker. Only seems to affect the bundle upload with a couple of 100-300kb.

Can use this as a workaround in the meantime 
```
    "predeploy": [
      "turbo build",
      "isolate",
      "cd isolate && pnpm dedupe"
    ],
```

The use-case I had was copy some files from another workspace package. The "files" field and running the scripts before isolate. I packaged a standalone nextjs server and use that in the deployed function. But got it working and it seems to be running without a hitch :) 

Thanks for the quick response!

**Comment by 0x80** [comment]
@torbjorn-kvist This should be fixed in 1.12.0 and the latest version of firebase-tools-with-isolate.

I found a pnpm lockfilePrune function that does the job. I think dedupe is a little bit different in that it can also change versions, but the former seems the correct one to use. You are free to run dedupe on your project before isolating of course.

**Comment by torbjorn-kvist** [comment]
Thanks for the quick fix! The lock file is way smaller now. The diff between running dedupe after is just a couple of bytes in my case, so I can skip that step now. 

Thanks again for finding the problem and fixing it :)
